prototype-spec.md

```diff
@@ -0,0 +1,253 @@
+   * `succinct_hash_delegations` :: A dictionary containing succinct descriptions of delegations to a set of bins, similar to those used in `path_hash_prefixes`. This field represents delegations to 2^BIT_LENGTH bins that use the keyid, threshold, path, and termination status from this delegation. The path_hash_prefixes and name for each bin will be determined using the BIT_LENGTH and NAME_PREFIX. BIT_LENGTH must be an integer between 1 and 32 (inclusive). The rolename for each bin will be structured as NAME_PREFIX-COUNT where COUNT is a hexadecimal value between 0 and 2^BIT_LENGTH-1 (inclusive) that represents the bin number. If the succinct_hash_delegations field is present in a delegation, the name field will not be used as a rolename, and so is not required.
+   * `paths` :: A list of strings, where each string describes a path that the role is trusted to provide. Clients MUST check that a target is in one of the trusted paths of all roles in a delegation chain, not just in a trusted path of the role that describes the target file. PATHPATTERN can include shell-style wildcards and supports the Unix filename pattern matching convention. Its format may either indicate a path to a single file, or to multiple paths with the use of shell-style wildcards. For example, the path pattern "targets/*.tgz" would match file paths "targets/foo.tgz" and "targets/bar.tgz", but not "targets/foo.txt". Likewise, path pattern "foo-version-?.tgz" matches "foo-version-2.tgz" and "foo-version-a.tgz", but not "foo-version-alpha.tgz". To avoid surprising behavior when matching targets with PATHPATTERN, it is RECOMMENDED that PATHPATTERN uses the forward slash (/) as directory separator and does not start with a directory separator, akin to TARGETPATH.
+
+Prioritized delegations allow clients to resolve conflicts between delegated roles that share responsibility for overlapping target paths. To resolve conflicts, clients must consider metadata in order of appearance of delegations; we treat the order of delegations such that the first delegation is trusted over the second one, the second delegation is trusted more than the third one, and so on. Likewise, the metadata of the first delegation will override that of the second delegation, the metadata of the second delegation will override that of the third one, etc. In order to accommodate prioritized delegations, the "roles" key in the DELEGATIONS object above points to an array of delegated roles, rather than to a hash table.
```

Comment:
To resolve conflicts, clients must consider metadata in the order of appearance of delegations, meaning the first delegation is trusted over the second one, the second delegation is trusted more than the third one, and so on.